scripts/xtensa-build-zephyr.py

```diff
@@ -0,0 +1,432 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: BSD-3-Clause
+
+import argparse
+import shlex
+import subprocess
+import pathlib
+import platform
+import sys
+import shutil
+import multiprocessing
+import os
+import warnings
+# anytree module is defined in Zephyr build requirements
+from anytree import Node, RenderTree
+
+# Constant value resolves SOF_TOP directory as: "this script directory/.."
+SOF_TOP = pathlib.Path(__file__).parents[1].resolve()
+# Default value may be overriten by -p arg
+west_top = pathlib.Path(SOF_TOP, "zephyrproject")
+default_rimage_key = pathlib.Path("modules", "audio", "sof", "keys", "otc_private_key.pem")
+
+if platform.system() == "Windows":
+	xtensa_tools_version_postfix = "-win32"
+elif platform.system() == "Linux":
+	xtensa_tools_version_postfix = "-linux"
+else:
+	warnings.warn(f"Your operating system: {platform.system()} is not supported")
+
+platform_list = [
+	# Intel platforms
+	{
+		"name": "apl",
+		"PLAT_CONFIG": "intel_adsp_cavs15",
+		"XTENSA_CORE": "X4H3I16w2D48w3a_2017_8",
+		"XTENSA_TOOLS_VERSION": f"RG-2017.8{xtensa_tools_version_postfix}"
+	},
+	{
+		"name": "cnl",
+		"PLAT_CONFIG": "intel_adsp_cavs18",
+		"XTENSA_CORE": "X6H3CNL_2017_8",
+		"XTENSA_TOOLS_VERSION": f"RG-2017.8{xtensa_tools_version_postfix}"
+	},
+	{
+		"name": "icl",
+		"PLAT_CONFIG": "intel_adsp_cavs20",
+		"XTENSA_CORE": "X6H3CNL_2017_8",
+		"XTENSA_TOOLS_VERSION": f"RG-2017.8{xtensa_tools_version_postfix}"
+	},
+	{
+		"name": "tgl",
+		"PLAT_CONFIG": "intel_adsp_cavs25",
+		"XTENSA_CORE": "cavs2x_LX6HiFi3_2017_8",
+		"XTENSA_TOOLS_VERSION": f"RG-2017.8{xtensa_tools_version_postfix}",
+		"RIMAGE_KEY": pathlib.Path("modules", "audio", "sof", "keys", "otc_private_key_3k.pem")
+	},
+	{
+		"name": "tgl-h",
+		"PLAT_CONFIG": "intel_adsp_cavs25",
+		"XTENSA_CORE": "cavs2x_LX6HiFi3_2017_8",
+		"XTENSA_TOOLS_VERSION": f"RG-2017.8{xtensa_tools_version_postfix}",
+		"RIMAGE_KEY": pathlib.Path("modules", "audio", "sof", "keys", "otc_private_key_3k.pem")
+	},
+	# NXP platforms
+	{
+		"name": "imx8",
+		"PLAT_CONFIG": "nxp_adsp_imx8",
+		"RIMAGE_KEY": "" # no key needed for imx8
```

Comment:
Why the inconsistency between empty/not needed vs ignored between the NXP platforms? There's nothing intrinsically different between imx8 and imx8x platforms.

Is this inconsistency in the original script?

Reply:
Guess that's fine then, I'd have to come and fix it later myself then.